### PR TITLE
Assignment-4: Promises in JavaScript

### PR DESCRIPTION
**Pull request by vaibhav-webonise**
**problem 1:** Addition using promise
**problem 2:** Division using promise
**problem 3:** Delay operation using promise
**promise 4:** Print numbers after some interval using promise

### CODE REVIEW

File: AdditionPromise.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
```
Comment (deepakb-webonise) [inline AdditionPromise.html:4]: The assignment looks good. try to use the fat arrow functions

File: AdditionPromise.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        <title>Addition</title>
+        <script>
+            function getData(){
+                let numberOne=parseInt(document.getElementById('numberOne').value);
+                let numberTwo=parseInt(document.getElementById('numberTwo').value);
+
+                let additionResult = addTwoNumbers(numberOne,numberTwo);
```
Comment (deepakb-webonise) [inline AdditionPromise.html]: no need additionResult variable we can directly then on function.

File: DivisionPromise.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Division</title>
+        <script>
```
Comment (deepakb-webonise) [inline DivisionPromise.html:6]: The assignment looks good. checkout the previous assignment comments same applies here

**Review by deepakb-webonise — APPROVED** [review]

File: DivisionPromise.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            function getData() {
+                let numberOne = parseInt(document.getElementById('numberOne').value);
```
Comment (swapnil-webonise) [inline DivisionPromise.html]: Prefer `const` over `let`, as we are not modifying a value further

Comment (swapnil-webonise) [reply DivisionPromise.html]: Check this across PR

File: PrintNumbersbyPromise.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        <script>
+            function performOperation() {
+                let numberByUser = document.getElementById('numberByUser').value;
+                displayNumbers(numberByUser).then(() => {
+                    document.write("starting numbers printing...");
+                }).catch(() => {
+                    document.write('Error while printing numbers...');
+                })
+            }
+            function displayNumbers(number) {
```
Comment (swapnil-webonise) [inline PrintNumbersbyPromise.html:16]: Give linebreak between two functions

**Review by swapnil-webonise — CHANGES_REQUESTED** [review]

**Review by swapnil-webonise — APPROVED** [review]

**Pull request merged**